tsfresh/feature_extraction/extraction.py

```diff
@@ -76,23 +85,25 @@ def extract_features(timeseries_container, feature_extraction_settings=None,
         for key in kind_to_df_map:
             feature_extraction_settings.set_default_parameters(key)
 
+    # Choose the parallelization according to a rule-of-thumb
+    if parallelization is None and (feature_extraction_settings.n_processes / 2) > len(kind_to_df_map):
```

Comment:
What about: using per Kind as default if number of kind bigger than 1 otherwise by sample?